### ISSUE TITLE
PirateBoar Bug

### ISSUE THREAD

**Issue by Russell2023** [atlasgames-repo/fortress-defense#107]
One of the enemies called pirate boar has several errors after its death.

Error1:
'[ Enemy ] PirateBoar - Range(Clone)' AnimationEvent 'SetHead' on animation 'Death' has no receiver! Are you missing a component?

Error2:
NullReferenceException: Object reference not set to an instance of an object
SmartEnemyGrounded.CheckAttack () (at Assets/_FD/Script/SmartEnemy/SmartEnemyGrounded.cs:219)
SmartEnemyGrounded.LateUpdate () (at Assets/_FD/Script/SmartEnemy/SmartEnemyGrounded.cs:146)

**Comment by m-nt** [comment]
In file `Assets/_FD/Script/SmartEnemy/SmartEnemyGrounded.cs:219` as followed :
``` c#
//CHECK AND CALL MINION IF THIS ENEMY HAS SCRIPT CALLMINION
    switch (attackType) {
        case ATTACKTYPE.RANGE:
            if (rangeAttack.AllowAction())
```
its checking for enemy atack type, check enemy prefab for Attack type component (range attack type)

**Comment by Russell2023** [comment]
but the code in cs:219 is in this format:

if (meleeAttack.AllowAction())
                {
                    if (meleeAttack.CheckPlayer(isFacingRight()))
                    {
                        SetEnemyState(ENEMYSTATE.ATTACK);
                        meleeAttack.Action();
                        AnimSetTrigger("melee");
                        SetSkeletonAnimation(ANIMATION_STATE.ATTACK, true);
                    }

**Comment by Russell2023** [comment]
are you talking about the Throw Attack script?

**Comment by m-nt** [comment]
Yes I know, the error (`NullReferenceException`) referes to the attack type component

**Comment by Russell2023** [comment]
ok I checked the script. should I change something?

**Comment by Russell2023** [comment]
everything looks fine except the death sound which is missing.

**Comment by m-nt** [comment]
@Russell2023 any results ?

**Comment by Russell2023** [comment]
No results till now. I've tried several methods and none of them seem to work. Here is a picture of that cursed event.

![Uploading fortress-defense - Playing atlas Test - PC, Mac & Linux Standalone - Unity 2020.2.5f1 Personal _DX11_ 1_13_2024 12_23_52 PM.png…]()

**Comment by m-nt** [comment]
fixed in https://github.com/atlasgames-repo/fortress-defense/pull/108
